source/Meadow.Core/Cloud/MeadowCloudConnectionService.cs

```diff
@@ -633,29 +634,20 @@ private async Task Send<T>(T item, string endpoint)
             }
 
             Resolver.Log.Debug($"making cloud request to {endpoint} with payload: {json}", "cloud");
-
             HttpResponseMessage response = await client.PostAsync($"{endpoint}", content);
-
-            if (response.IsSuccessStatusCode)
-            {
-                Resolver.Log.Debug($"cloud request to {endpoint} completed successfully", messageGroup: "cloud");
-            }
-            else if (response.StatusCode == HttpStatusCode.Unauthorized)
+            Resolver.Log.Debug($"{response.StatusCode.ToString()}", messageGroup: "cloud");
+            
+            if (response.StatusCode == HttpStatusCode.Unauthorized && attempt < maxRetries)
             {
-                if (attempt < maxRetries)
-                {
-                    attempt++;
-                    Resolver.Log.Debug($"cloud request to {endpoint} unauthorized, re-authenticating", "cloud");
-                    // by setting this to null and retrying, Authenticate will get called
-                    ClientOptions = null;
-                    _jwt = null;
-                    goto retry;
-                }
-
-                errorMessage = $"cloud request to {endpoint} failed with {response.StatusCode}";
-                throw new MeadowCloudException(errorMessage);
+                attempt++;
+                // by setting this to null and retrying, Authenticate will get called
+                ClientOptions = null;
+                _jwt = null;
+                client = new HttpClient();
```

Comment:
The client created at line 604 above potentially is not disposed here

Reply:
good catch. I fixed and tested.